auth/data-structures.md

```diff
@@ -0,0 +1,92 @@
+# Data Structures
+
+## Request Params
+
+```typescript
+interface RequestParams {
+  chainId: string;
+  nonce: string;
+  aud?: string;
+  nbf?: string;
+  exp?: string;
+  statement?: string;
+  requestId?: string;
+  resources?: string[];
+}
```

Comment:
```suggestion
interface RequestParams {
  domain: string;
  chainId: string;
  nonce: string;
  aud: string;
  nbf?: string;
  exp?: string;
  statement?: string;
  requestId?: string;
  resources?: string[];
}
```

Reply:
from specs:
uri in siwe is not optional so aud should not be as well
`URI: {.p.aud}`
domain is also not optional